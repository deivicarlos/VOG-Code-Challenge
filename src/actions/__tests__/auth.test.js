import configureMockStore from 'redux-mock-store';
import {
  requestLoginBegin,
  requestLoginSuccess,
  requestLoginError,
} from '../auth';

const mockStore = configureMockStore();
const store = mockStore({});

describe('Auth action creators', () => {
  it('creates LOGIN_REQUEST_BEGIN before making API call', () => {
    // Dispatch the createSuccess action with the values of a new to-do.
    store.dispatch(requestLoginBegin());
    expect(store.getActions()).toMatchSnapshot();
  });

  it('creates LOGIN_REQUEST_SUCCESS when api call is made successfully', () => {
    store.dispatch(requestLoginSuccess());
    expect(store.getActions()).toMatchSnapshot();
  });

  it('creates LOGIN_REQUEST_ERROR when API call fails', () => {
    // Dispatch the createSuccess action with the values of a new to-do.
    store.dispatch(requestLoginError());
    expect(store.getActions()).toMatchSnapshot();
  });
});
