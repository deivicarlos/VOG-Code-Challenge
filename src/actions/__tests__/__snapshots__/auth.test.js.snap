// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth action creators creates LOGIN_REQUEST_BEGIN before making API call 1`] = `
Array [
  Object {
    "type": "LOGIN_REQUEST_BEGIN",
  },
]
`;

exports[`Auth action creators creates LOGIN_REQUEST_ERROR when API call fails 1`] = `
Array [
  Object {
    "type": "LOGIN_REQUEST_BEGIN",
  },
  Object {
    "type": "LOGIN_REQUEST_SUCCESS",
  },
  Object {
    "type": "LOGIN_REQUEST_ERROR",
  },
]
`;

exports[`Auth action creators creates LOGIN_REQUEST_SUCCESS when api call is made successfully 1`] = `
Array [
  Object {
    "type": "LOGIN_REQUEST_BEGIN",
  },
  Object {
    "type": "LOGIN_REQUEST_SUCCESS",
  },
]
`;
